Restructure dashboard sharing page to include export and embed info (#5592)
elastic/docs-content: explore-analyze/dashboards/sharing.md

---
mapped_pages:
  - https://www.elastic.co/guide/en/kibana/current/share-the-dashboard.html
description: Share Kibana dashboards with your team using links or embeds, and export them as PDF, PNG, CSV, JSON, or NDJSON files.
applies_to:
  stack: ga
  serverless: ga
products:
  - id: kibana
---

# Share and export dashboards [share-the-dashboard]

{{kib}} provides several ways to share and export dashboards:

- [Share with a link](#share-dashboard-link): Copy a direct link to the dashboard.
- [Embed in a webpage](#embed-dashboard): Embed an interactive dashboard as an iframe on external web pages.
- [Export as PDF or PNG](#export-dashboard-pdf-png): Generate a report file of your dashboard.

> - [Download visualization data as CSV](#download-csv): Download the data behind individual Lens visualizations.

Author: ```suggestion
- [Download visualization data as CSV](#download-csv): Download the data behind individual visualizations.
```

- [Export dashboard configuration](#export-dashboards): Export as NDJSON for backup or migration, or as JSON for the dashboards API.

> You can also [set who can edit the dashboard](#manage-dashboard-access).

Author: ```suggestion
You can also [set whether other users of your space can edit dashboards you own](#manage-dashboard-access).
```

> Most options start from the {icon}`share` **Share** button in the dashboard application menu, where you can also [set who can edit the dashboard](#manage-dashboard-access). CSV download and dashboard configuration export use different entry points; see each section for details.

Author: ```suggestion
You can also [set who can edit the dashboard](#manage-dashboard-access).
```

## Embed in a webpage [embed-dashboard]
```{applies_to}
stack: ga
serverless: unavailable
```

Embed a fully interactive dashboard as an iframe on an internal company website or personal web page.

1. Open the dashboard, then select {icon}`share` **Share** in the application menu.
2. Go to the **Embed** tab.
3. Under **Include**, select which parts of the dashboard to display in the embedded view:
   - **Top menu**: the menu bar at the top of the dashboard
   - **Query**: the search query bar
   - **Time filter**: the time range selector
   - **Filter bar** (enabled by default): the bar showing applied filters
4. Select **Copy embed code**.
5. Paste the iframe code into your web page HTML.

Save the dashboard before you generate the embed code. Embedding a dashboard with unsaved changes might result in an embed code that does not work properly.

For information about granting access to embedded dashboards, refer to [Authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/user-authentication.md).

## Export as PDF or PNG [export-dashboard-pdf-png]
```{applies_to}
stack: ga
serverless: unavailable
```

> Generate and download a PDF or PNG file of your dashboard. PDF and PNG reports are a [subscription feature](https://www.elastic.co/subscriptions).

Reviewer: Would it be worth being a little more detailed on what a report might contain here? To me, it's not quite as intuitive as some of the other actions.

Author: Yeah I added a bit of details, good point, thank you!

1. Open the dashboard, then select {icon}`download` **Export** in the application menu.
2. Choose the PDF or PNG option.
3. Optional: For PDF exports, enable **Print format** to create a printer-friendly report with multiple A4 portrait pages and two visualizations per page.
4. Select **Export PDF** or **Export PNG** to generate the report.

A notification confirms that the report is queued. When it is ready, download it from the **Reporting** page under **Stack Management** → **Alerts and Insights** → **Reporting**. If the report contains a data table or Discover session, the PDF includes only the visible data.

{applies_to}`stack: ga 9.1+` You can also schedule recurring exports. Refer to [Automatically generate reports](../report-and-share/automating-report-generation.md) to learn more.

For general information about reporting across all {{kib}} apps, including known limitations and troubleshooting, refer to [Reporting and sharing](../report-and-share.md). To set up reporting, see [Configure reporting](/deploy-manage/kibana-reporting-configuration.md).

> ## Download visualization data as CSV [download-csv]

Reviewer: Is there a difference between "Export as PDF" and "Download...as CSV" in these options or just matching the UI? The docs also use Export for the config, which seems slightly different, but maybe there's nothing you can do with the language in the UI.

Author: yeah theres's a mix of UI language here + these actions don't necessarily all touch the same things, CSV downloads gets you the data of some panels, PDF/PNG exports get you a screenshot basically. I made improvements to the page so that it's clearer, thanks for the comment!

   * Select **Download JSON** to save the JSON source to a file.

::::

::::{applies-item} {stack: ga 9.0-9.3}

Export dashboards as NDJSON files to migrate them to other {{kib}} instances, back them up, or share them with other teams. You can export dashboards from **Stack Management** → **Saved Objects**. To configure and start the export:

1. Select the dashboard that you want, then select **Export**.
2. Enable **Include related objects** if you want objects associated with the selected dashboard, such as data views and visualizations, to also be exported. This option is enabled by default and recommended if you plan to import that dashboard again in a different space or cluster.
3. Select **Export**.

To automate {{kib}}, you can export dashboards as NDJSON using the [Export saved objects API]({{kib-apis}}group/endpoint-saved-objects). It is important to export dashboards with all necessary references.

:::{note}
You can still use the NDJSON export on later versions, but the API-compatible JSON export is the recommended way to manage a single dashboard as code.
:::

::::

:::::

> ## Set edit permissions [manage-dashboard-access]

Reviewer: Do you need edit permissions to share/download/export? Maybe we could note that as well if that's the case.

Author: Good point, I added some information in the relevant sections and when there are actually permissions involved (and a subscription level for the case of reporting).

::::

::::{applies-item} {stack: ga 9.0-9.3}

Export dashboards as NDJSON files to migrate them to other {{kib}} instances, back them up, or share them with other teams. You can export dashboards from **Stack Management** → **Saved Objects**. To configure and start the export:

1. Select the dashboard that you want, then select **Export**.
2. Enable **Include related objects** if you want objects associated with the selected dashboard, such as data views and visualizations, to also be exported. This option is enabled by default and recommended if you plan to import that dashboard again in a different space or cluster.
3. Select **Export**.

To automate {{kib}}, you can export dashboards as NDJSON using the [Export saved objects API]({{kib-apis}}group/endpoint-saved-objects). It is important to export dashboards with all necessary references.

:::{note}
You can still use the NDJSON export on later versions, but the API-compatible JSON export is the recommended way to manage a single dashboard as code.
:::

::::

:::::

## Set edit permissions [manage-dashboard-access]
```{applies_to}
serverless: ga
stack: ga 9.3+
```

> Only the dashboard owner or a {{kib}} administrator can control whether other users in the space can edit the dashboard or only view it. To set this, open the dashboard, select {icon}`share` **Share** in the application menu, and choose:

Author: ```suggestion
As the owner of a dashboard or {{kib}} administrator, you can control whether other users in the space can edit it or only view it. To set this, open the dashboard, select {icon}`share` **Share** in the application menu, and choose:
```
